qcodes/instrument_drivers/tektronix/AWG70000A.py

```diff
@@ -830,12 +830,9 @@ def _makeWFMXFileBinaryData(data: np.ndarray, amplitude: float) -> bytes:
             N = shape[1]
             M = shape[0]
             wfm = data[0, :]
-            if M == 2:
-                markers = data[1, :]
-            elif M == 3:
-                m1 = data[1, :]
-                m2 = data[2, :]
-                markers = m1+2*m2  # This is how one byte encodes both markers
+            markers = data[1, :]
+            for i in range(1, M-1):
+                markers += data[i+1, :] * (2**i)
                 markers = markers.astype(int)
```

Comment:
`markers = markers.astype(int)` has to be outside loop. Otherwise it throws an error when it tries to add float number to int.